src/plugins/dataflow/components/dataflow-program.tsx

```diff
@@ -514,10 +522,24 @@ export class DataflowProgram extends BaseComponent<IProps, IState> {
   };
 
   private tick = () => {
-    // Update the sampling rate
     const now = Date.now();
     this.setState({lastIntervalDuration: now - this.lastIntervalTime});
     this.lastIntervalTime = now;
+    const isRecording = this.props.programRecordState === 1;
+
+    /* ==[ Per tick - create a case and write it to the dataSet ] == */
+    if (isRecording){
+      const aCase = {
+        __id__: newCaseId(),
+      };
+      const dataSet = this.props.tileModel.dataSet;
+      //loop through attribute (nodes) and write each value
+      this.programEditor.nodes.forEach((node, idx) => {
+        const key = dataSet.attributes[idx].id as keyof typeof aCase;
```

Comment:
```suggestion
        const key = dataSet.attributes[idx].id;
```
You don't need the `as` type cast if `aCase` is typed appropriately.